include/tester/vtcp/tcp/TCPSession.hpp

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include<memory>
+#include<functional>
+#include<atomic>
+#include<string>
+
+#include<boost/asio.hpp>
+
+#include"TCPReceiver.hpp"
+#include"TCPTransmitter.hpp"
+#include"../generic/Message.hpp"
+
+using  boost::asio::ip::tcp;
+
+class TCPSession
+{
+public:
+    TCPSession(tcp::socket socket, const int id,
+    std::function<void(const int, const Message&)> onMessageReceived,
+    std::function<void(const int)> onDisconnect) : 
+        _socket(std::make_shared<tcp::socket>(std::move(socket))),
+        _active(std::make_shared<std::atomic<bool>>(false)),
+        _onMessageReceived(onMessageReceived),
+        _onDisconnect(onDisconnect),
+        _receiver(_socket, _active,
+            std::bind(&TCPSession::onDataReceived, this, std::placeholders::_1),
+            std::bind(&TCPSession::onDisconnect, this)),
+        _transmitter(_socket, _active,
+            std::bind(&TCPSession::onDisconnect, this))
+    {
+        _id = id;
+    }
+
+    ~TCPSession() = default;
+
+    void start()
+    {
+        if(!*_active)
+        {
+            *_active = true;
+            _receiver.run();
+        }
+    }
+
+    void send(const std::vector<std::uint8_t> & message)
+    {
+        _transmitter.write(message);
+    }
+
+    void onDisconnect()
```

Comment:
Put this method under the `private` block, and add another public method named `disconnect` that calls this `onDisconnect` method. the public `disconnect` would be useful in `VTCPRequestHandler::onVtcpDisconnect`

Reply:
done

Reply:
Which function should I send now to the Receiver and the Transmitter `disconnect` or `onDisconnect`?

Reply:
the private `onDisconnect`. the public `disconnect` shall be used whenever you want to close the session not via callbacks of the receiver / transmitter (Such as the VTCPRequestHandler)

Reply:
ok